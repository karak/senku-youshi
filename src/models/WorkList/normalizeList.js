import map from 'lodash/map';
import split from 'lodash/split';
import filter from 'lodash/filter';
import trim from 'lodash/trim';

/**
 * normalize list of works with noise
 *
 * @param {string} value - original value
 * @returns {string} - normalized value
 *
 * @todo implementation
 */
export default function normalizeList(value) {
  return split(value, /\r?\n/)
    .map(x => trim(x))
    .filter((x, i, xs) => x.length > 0 || i === xs.length - 1)
    .join('\n');
}
