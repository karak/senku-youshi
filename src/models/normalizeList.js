/**
 * normalize list of works with noise
 *
 * @param {string} value - original value
 * @returns {string} - normalized value
 * 
 * @todo implementation
 */
export default function normalizeList(value) {
  return value;
}
