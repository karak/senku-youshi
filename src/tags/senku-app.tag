<senku-app>
  <command-bar ref="commandBar" />
  <controlled-textarea ref="editor" />
  <script>
  import './controlled-textarea.tag';
  import './command-bar.tag';
  import normalizeList from '../models/normalizeList';
  import shuffle from '../models/shuffle';

  this.editorValue = '';
  this.on('mount', () => {
    this.refs.editor.on('change', ({ value }) => {
      this.refs.editor.value = normalizeList(value);
    });

    this.refs.commandBar.on('shuffle', () => {
      this.refs.editor.value = shuffle(this.refs.editor.value);
    });
  });
  </script>
</senku-app>
