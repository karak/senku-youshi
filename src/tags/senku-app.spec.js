import { shallow } from 'riot-test-utils';
import './senku-app.tag';

// mock normalize function
jest.mock('../models/normalizeList');
import normalizeList from '../models/normalizeList';
normalizeList.mockImplementation(x => x + '!'); // append "!"

describe('<senku-app />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow('senku-app');
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('has "editor" in refs', () => {
    expect(wrapper.instance().refs).toHaveProperty('editor');
  });

  it('normalizes the value on <senku-editor> changes its value', () => {
    const editor = wrapper.instance().refs.editor;

    editor.trigger('change', { value: 'Hello' });

    expect(editor.value).toBe('Hello' + '!');
  });
});
