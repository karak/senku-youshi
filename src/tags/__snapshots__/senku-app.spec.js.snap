// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<senku-app /> matches snapshot 1`] = `
Object {
  "attributes": Object {
    "data-is": Array [
      "senku-app",
    ],
  },
  "children": Array [
    Object {
      "attributes": Object {
        "ref": Array [
          "commandBar",
        ],
      },
      "children": Array [],
      "name": "command-bar",
    },
    " ",
    Object {
      "attributes": Object {
        "ref": Array [
          "editor",
        ],
      },
      "children": Array [],
      "name": "controlled-textarea",
    },
  ],
  "name": "senku-app",
}
`;
